pkg/cmd/extension/manager.go

```diff
@@ -172,7 +179,80 @@ func (m *Manager) InstallLocal(dir string) error {
 	return makeSymlink(dir, targetLink)
 }
 
-func (m *Manager) Install(cloneURL string, stdout, stderr io.Writer) error {
+type BinManifest struct {
+	Owner string
+	Name  string
+	Host  string
+	// TODO I may end up not using this; just thinking ahead to local installs
+	Path string
+}
+
+func (m *Manager) InstallBin(client *http.Client, repo ghrepo.Interface) error {
+	var r *release
+	r, err := fetchLatestRelease(client, repo)
+	if err != nil {
+		return err
+	}
+
+	suffix := m.platform()
+	var asset *releaseAsset
+	for _, a := range r.Assets {
+		if strings.HasSuffix(a.Name, suffix) {
```

Comment:
Is the condition for a binary attachment only that it ends with `GOOS-GOARCH` as suffix? Seems straightforward enough 👍 But other than the suffix, is there a specific naming pattern for assets?

Reply:
I was only planning on enforcing the suffix of GOOS-GOARCH; I'm open to more restrictions but didn't immediately see value in it.